Library/Homebrew/software_spec.rb

```diff
@@ -402,20 +404,37 @@ def compatible_locations?
   # Does the {Bottle} this {BottleSpecification} belongs to need to be relocated?
   sig { returns(T::Boolean) }
   def skip_relocation?
-    cellar == :any_skip_relocation
+    legacy_cellar_or_cellar_for_tag == :any_skip_relocation
   end
 
+  sig { params(tag: Symbol).returns(T::Boolean) }
   def tag?(tag)
     checksum_for(tag) ? true : false
   end
 
-  # Checksum methods in the DSL's bottle block optionally take
+  # BottleChecksum methods in the DSL's bottle block take
   # a Hash, which indicates the platform the checksum applies on.
-  def sha256(val)
-    digest, tag = val.shift
-    collector[tag] = Checksum.new(digest)
-  end
-
+  # Example args:
+  # {"22948764d8f8d7be4870ff92dae64d986eb63a9150b219c20fff87d1a6aa93d6"=>
+  # {"tag"=>:mojave, "cellar"=>:any_skip_relocation}}
+  def sha256(sha256_line_hash)
+    checksum_hash = sha256_line_hash.select { |key, _value| key.length == 64 }
+    checksum_value = checksum_hash.keys.first
+    tag = checksum_hash[checksum_value]
+    # Legacy DSL: There is only a single cellar line for all the tags
+    # New DSL: get the cellar for the specific tag
+    cellar_for_tag = cellar.presence || sha256_line_hash["cellar"]
+    cellar_for_tag ||= Homebrew::DEFAULT_CELLAR
+    collector[tag] = BottleChecksum.new(checksum_value, tag: tag, cellar: cellar_for_tag)
+  end
+
+  def legacy_cellar_or_cellar_for_tag
```

Comment:
I'd be tempted to say this should just be called `cellar` as it'll make the other code easier to follow as a result.

Reply:
This method is gone now.